pkg/provisioner/ironic/update_opts.go

```diff
@@ -72,6 +73,27 @@ func deref(v interface{}) interface{} {
 	return ptrVal.Elem().Interface()
 }
 
+func sanitisedValue(data interface{}) interface{} {
+	dataType := reflect.TypeOf(data)
+	if dataType.Kind() != reflect.Map ||
+		dataType.Key().Kind() != reflect.String {
+		return data
+	}
+
```

Comment:
The logic is faulty.  It will always return the original data because of the ||.

If the comment from Zane is correct, then it should only check if it's not a Map, return the original data.

Make sure the test case is working as expected.

Reply:
I'm not clear on what the fault in the logic is? It won't return the original data (here) if the data type is a Map and the map's keys are Strings. The test case proves that this is working.

Reply:
I realized it was dataType.Kind() and dataType.Key().Kind() (not the same) so it is okay.  Thanks.